enumerable.rb

```diff
@@ -0,0 +1,102 @@
+# Custom Enumerable Methods
+module ::Enumerable
+  # Calls block with the item for each item in enum.
+  def my_each
+    return to_enum unless block_given?
+
+    length.times { |n| yield self[n] }
+    self
+  end
+
+  # Loops, Calling The Block With Item And Index
+  def my_each_with_index(&block)
+    for index in 0...count
+      block.call(self[index], index)
+    end
+  end
+
+  # Returns an array containing all elements of enum for which the given block returns a true value.
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each do |n|
+      array << n if yield n
+    end
+    array
+  end
```

Comment:
### #my_each_with_index

your #my_select is having some issue when is tested with a range, before testing this issue make sure you fix the #my_each method.

Reply:
Fixed in https://github.com/epinczinger/ruby_enumerables/pull/7/commits/0b50271022ec8e7bfa23f47e627e8ca5e0cdf7d2